Daemon/Daemon/Monitor.m

```diff
@@ -705,19 +706,13 @@ -(PluginBase*)findPlugin:(File*)file
     __block PluginBase* plugin = nil;
     
     //look for plugin that can handle path
-    [plugins enumerateObjectsUsingBlock:^(PluginBase* currentPlugin, NSUInteger index, BOOL* stop)
```

Comment:
`enumerateObjectsUsingBlock:` was blocking current thread while enumerating through the collection of the registered plugins sequentially. `indexOfObjectWithOptions:NSEnumerationConcurrent` will block the current thread, but also would fire multiple threads at the same time for looking up the needed plugin.

Might work, might not work. For your consideration.